### PR TITLE
Revise upgrade and migration guides

### PR DESCRIPTION
**Pull request by jathanism**
### Fixes: #262 

- Revised the upgrading guide to reference correct steps for Nautobot
- Migration guide greatly expanded to included specific code migration instructions for the data model changes
- Renamed `nautobot.ipam.querysets.NetworkQueryset.net_contains_or_equal` to `net_contains_or_equals` for parity from network field lookup names from NetBox and because of how it reads ("equals" is grammatically correct, and this is the time to do it)

### CODE REVIEW

**Comment by jathanism** [comment]
@mzbroch It would be great to get your feedback on the migration guide changes, especially for the IPAM model changes.

File: nautobot/docs/installation/migrating-from-netbox.md [hunk 1]
```diff
@@ -81,3 +116,399 @@ Database keys are now defined as randomly-generated [Universally Unique Identifi
...
+
+All IPAM objects with network field types (`ipam.Aggregate`, `ipam.IPAddress`, and `ipam.Prefix`) are no longer hard-coded to use PostgreSQL-only `inet` or `cidr` field types and are now using a custom implementation leveraging SQL-standard `varbinary` field types.
+
+#### Technical Details
+
+Below is a summary of the underlying technical changes to network fields. These will be explained in more detail in the following sections.
+
+- For `IPAddress`, the `address` field was exploded out to `host`, `broadcast`, and `prefix_length` fields; `address` was converted into a computed field.
+- For `Aggregate` and `Prefix` objects, the `prefix` field was exploded out to `network`, `broadcast`, and `prefix_length` fields; `prefix` was converted into a computed field.
+- The `host`, `network`, and `broadcast` fields are now of a `varbinary` database type, which is represented as a packed binary integer (for example, the string value`"1.1.1.1"` is packed as `b"\x01\x01\x01\x01"`)
```
Comment (glennmatthews) [inline nautobot/docs/installation/migrating-from-netbox.md:143]:
```suggestion
- The `host`, `network`, and `broadcast` fields are now of a `varbinary` database type, which is represented as a packed binary integer (for example, the host `1.1.1.1` is packed as `b"\x01\x01\x01\x01"`)
```

File: nautobot/docs/installation/migrating-from-netbox.md [hunk 2]
```diff
@@ -81,3 +116,399 @@ Database keys are now defined as randomly-generated [Universally Unique Identifi
...
+
+Nautobot:
+
+```python
+IPAddress.objects.net_in([n1, n2, n3])
+```
+
+##### ip_family
+
+*Returns target addresses matching the source addresses family.*
```
Comment (glennmatthews) [inline nautobot/docs/installation/migrating-from-netbox.md:473]:
```suggestion
*Returns target addresses matching the specified family.*
```

File: nautobot/docs/installation/upgrading.md [hunk 3]
```diff
@@ -19,92 +19,51 @@ Nautobot v1.0.0 and later requires the following:
...
 
-This guide assumes that Nautobot is installed at `/opt/nautobot`. Pull down the most recent iteration of the `main`
-branch:
+Then, upgrade your dependencies using `pip3`:
 
 ```no-highlight
-cd /opt/nautobot
-sudo git checkout main
-sudo git pull origin main
+$ pip3 install --upgrade $NAUTOBOT_ROOT/local_requirements.txt
```
Comment (glennmatthews) [inline nautobot/docs/installation/upgrading.md:43]: Perhaps?

```suggestion
$ pip3 install --upgrade -r $NAUTOBOT_ROOT/local_requirements.txt
```

**Review by glennmatthews — COMMENTED** [review]
Nice work - thank you for the writeup!

File: nautobot/docs/installation/migrating-from-netbox.md [hunk 4]
```diff
@@ -81,3 +116,398 @@ Database keys are now defined as randomly-generated [Universally Unique Identifi
...
+
+```python
+>>> net.network
+'1.1.1.0'
+```
+
+This field *can* be used in nested filter expressions, for example:
+
+```python
+>>> Prefix.objects.filter(network="1.1.1.0")
```
Comment (mzbroch) [inline nautobot/docs/installation/migrating-from-netbox.md:287]: Can we show a nested filter in this example?

File: nautobot/docs/installation/migrating-from-netbox.md [hunk 5]
```diff
@@ -81,3 +116,398 @@ Database keys are now defined as randomly-generated [Universally Unique Identifi
...
+NetBox:
+
+```python
+Prefix.objects.filter(prefix__net_contained_or_equal=str(self.prefix))
+```
+
+Nautobot:
+
+```python
+Prefix.objects.net_contained_or_equal(self.prefix)
```
Comment (mzbroch) [inline nautobot/docs/installation/migrating-from-netbox.md:364]: `s` typo

**Comment by mzbroch** [comment]
Great stuff Jathan!

**Review by mzbroch — APPROVED** [review]

**Review by glennmatthews — APPROVED** [review]

**Pull request merged**